framework/core/js/src/common/states/ModalManagerState.ts

```diff
@@ -19,44 +25,57 @@ export default class ModalManagerState {
   /**
    * @internal
    */
-  modal: null | {
-    componentClass: UnsafeModalClass;
-    attrs?: Record<string, unknown>;
-    key: number;
-  } = null;
+  modal: ModalItem = null;
+
+  /**
+   * @internal
+   */
+  modalList: Array<ModalItem> = [];
 
   /**
    * Used to force re-initialization of modals if a modal
    * is replaced by another of the same type.
    */
   private key = 0;
 
-  private closeTimeout?: NodeJS.Timeout;
-
   /**
    * Shows a modal dialog.
    *
    * If a modal is already open, the existing one will close and the new modal will replace it.
    *
+   * Stacking modals is supported, the previous modal will stay behind the new modal.
+   *
    * @example <caption>Show a modal</caption>
    * app.modal.show(MyCoolModal, { attr: 'value' });
    *
    * @example <caption>Show a modal from a lifecycle method (`oncreate`, `view`, etc.)</caption>
    * // This "hack" is needed due to quirks with nested redraws in Mithril.
    * setTimeout(() => app.modal.show(MyCoolModal, { attr: 'value' }), 0);
+   *
+   * @example <caption>Stacking modals</caption>
+   * app.modal.show(MyCoolStackedModal, { attr: 'value' }, true);
    */
-  show(componentClass: UnsafeModalClass, attrs: Record<string, unknown> = {}): void {
+  show(componentClass: UnsafeModalClass, attrs: Record<string, unknown> = {}, stackModal: Boolean = false): void {
```

Comment:
We should use `boolean` primitive rather than the `Boolean` object.

```suggestion
  show(componentClass: UnsafeModalClass, attrs: Record<string, unknown> = {}, stackModal: boolean = false): void {
```